src/container/srv_target.c

```diff
@@ -99,21 +102,17 @@ cont_child_aggregate(struct ds_cont_child *cont, uint64_t *sleep)
 	if (rc)
 		return rc;
 
-	/*
-	 * sc_aggregation_min != DAOS_EPOCH_MAX means we need to aggregate
-	 * from epoch 0 because some snapshot was deleted.
-	 */
-	if (cont->sc_aggregation_min != DAOS_EPOCH_MAX) {
-		epoch_min = 0;
-		/*
-		 * Set sc_aggregation_min to non-zero so that we can tell
-		 * if another snapshot deletion happened when this round of
-		 * aggregation done.
+	change_hlc = max(cont->sc_snapshot_change_hlc,
+			 cont->sc_pool->spc_rebuild_fence);
```

Comment:
We need two fence_hlc per pool: 'spc_rebuild_fence' for current fence, 'spc_last_rebuild_fence' for the last finished rebuild, here we should compare 'full_scan_hlc' with the 'last_rebuild_fence'.